src/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/DiagnosticSourceEventSource.cs

```diff
@@ -2,32 +2,193 @@
+    ///
+    /// Example1:
+    /// 
+    ///    "BridgeTestSource1/TestEvent1:cls_Point_X=cls.Point.X;cls_Point_Y=cls.Point.Y\r\n" + 
+    ///    "BridgeTestSource2/TestEvent2:-cls.Url"
+    ///   
+    /// This indicates that two events should be turned on, The 'TestEvent1' event in BridgeTestSource1 and the
+    /// 'TestEvent2' in BridgeTestSource2.   In the first case, because the transform did not begin with a - 
+    /// any primitive type/string of 'TestEvent1's payload will be serialized into the output.  In addition if
+    /// there a property of the payload object called 'cls' which in turn has a property 'Point' which in turn
+    /// has a property 'X' then that data is also put in the output with the name cls_Point_X.   Simmilarly 
+    /// if cls.Point.Y exists, then that value will also be put in the output with the name cls_Point_Y.
+    /// 
+    /// For the 'BridgeTestSource2/TestEvent2' event, becasue the - was specified NO implicit fields will be 
+    /// generated, but if there is a property call 'cls' which has a property 'Url' then that will be placed in
+    /// the output with the name 'Url' (since that was the last property name used and no Variable= clause was 
+    /// specified. 
+    /// 
+    /// Example:
+    /// 
+    ///     "BridgeTestSource1\r\n" + 
+    ///     "BridgeTestSource2"
+    ///     
+    /// This will enable all events for the BridgeTestSource1 and BridgeTestSource2 sources.   Any string/primtive 
+    /// properties of any of the events will be serialized into the output.  
+    /// 
+    /// Example:
+    /// 
+    ///     ""
+    ///     
+    /// This turns on all DiagnosticSources Any string/primtive properties of any of the events will be serialized 
+    /// into the output.   This is not likely to be a good idea as it will be very verbose, but is useful to quickly
+    /// discover what is available.
+    /// 
+    /// See the DiagnosticSourceEventSourceBridgeTest.cs for more explicit examples of using this bridge.  
     /// </summary>
     [EventSource(Name = "Microsoft-Diagnostics-DiagnosticSource")]
     internal class DiagnosticSourceEventSource : EventSource
     {
         public static DiagnosticSourceEventSource Logger = new DiagnosticSourceEventSource();
 
+        class Keywords
+        {
+            /// <summary>
+            /// Indicates diagnostics messages from DiagnosticSourceEventSource should be included. 
+            /// </summary>
+            public const EventKeywords Messages = (EventKeywords)1;
+            /// <summary>
+            /// Indicates that all events from all diagnostic sources should be forwarded to the EventSource using the 'Event' event.  
+            /// </summary>
+            public const EventKeywords Events = (EventKeywords)2;
+        };
+
+        /// <summary>
+        /// Used to send ad-hoc diagnostics to humans.   
+        /// </summary>
+        [Event(1, Keywords = Keywords.Messages)]
+        public void Message(string Message)
```

Comment:
Yes.   The names of arguments end up being the names of properties when the event is parsed.   Thus one way or the the other our casing convention gets violated.   I choose the convention for properties, because that is probably how people think about them most of the time, and that is how you would do it if you use Write<T>( .. new { ...}).